src/functions-and-arrays.js

```diff
@@ -1,73 +1,603 @@
 // Iteration #1: Find the maximum
-
+function maxOfTwoNumbers(num1, num2) {
+    if (num1 > num2) {
+        return num1;
+    } else if (num2 > num1) {
+        return num2;
+    } else if (num1 === num2) {
+        return num1;
+    }
+}
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+const words = [
+    "mystery",
+    "brother",
+    "aviator",
+    "crocodile",
+    "pearl",
+    "orchard",
+    "crackpot",
+];
+
+function findLongestWord(arr) {
+    if (arr.length === 0) {
+        return null;
+    }
+    let longestWord = "";
+
+    for (let word of arr) {
+        if (word.length > longestWord.length) {
+            longestWord = word;
+        }
+    }
+    return longestWord;
+}
 
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumNumbers(arr) {
+    if (arr.length === 0) {
+        return 0;
+    }
+    let sumOfNumbers = 0;
+
+    for (let i = 0; i < arr.length; i++) {
+        sumOfNumbers += arr[i];
+    }
+    return sumOfNumbers;
+}
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(arr) {
+    if (arr.length === 0) {
+        return null;
+    }
+
+    let average = 0;
+    let sum = 0;
+
+    for (num of arr) {
+        sum += num;
+    }
```

Comment:
You could also have used the previously declared sumNumbers() function (line 40) here, using its value to calculate the averga!